### ISSUE TITLE
Improve error-handling around partially configured zone

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/omicron#215]
After chatting with @bnaecker , it seems like this is a relatively common pattern.

TL;DR: Someone is running Omicron on Helios, but oops, a dependency is missing. Stop, restart, re-run everything... except now there are leftover artifacts from the first time execution started.

In particular, the sled uses local storage for:
- A ZFS dataset for storing zones
- A base zone, intended to be cloned for propolis instances
- Child zones and NICs

We should have more rigor with these objects, in particular:
- Having some mechanism for verifying them if we observe them on a re-initialization of the sled agent, and
- Preferring to use static images which are verified where possible (e.g., could we have a hash-verified base zone instead of creating it on the fly? that seems better).
